config/application.rb

```diff
@@ -204,6 +200,13 @@ def try_config_for(*args)
     three_scale = config_for(:settings).symbolize_keys
     three_scale[:error_reporting_stages] = three_scale[:error_reporting_stages].to_s.split(/\W+/)
 
+    config.three_scale.payments = ActiveSupport::OrderedOptions.new
+    config.three_scale.payments.enabled = false
+    config.three_scale.payments.merge!(try_config_for(:payments) || {})
+    config.three_scale.payments.active_merchant_mode ||= three_scale.delete(:active_merchant_mode)&.to_sym || (Rails.env.production? ? :production : :test)
+    config.three_scale.payments.active_merchant_logging ||= three_scale.delete(:active_merchant_logging)
+    config.three_scale.payments.billing_canaries ||= three_scale.delete(:billing_canaries)
```

Comment:
In the case it already exist it won't delete the key in `three_scale` but I think that is OK

Or you could use `Hash#extract!`